content/docs/getting-started/hello-world.md

```diff
@@ -71,93 +56,74 @@ Next, we'll add some to the `client` `TcpStream`. This asynchronous task now cre
 future (in this case the stream will be created). This means that `TcpStream::connect` does
 not wait for the stream to be created before it returns. Rather it returns immediately
-with a value representing the work of creating a TCP stream. We'll see down below when this work
-_actually_ gets executed.
-
-The `and_then` method yields the stream once it has been created. `and_then` is an
-example of a combinator function that defines how asynchronous work will be processed.
-
-Each combinator function takes ownership of necessary state as well as the
-callback to perform and returns a new `Future` that has the additional "step"
-sequenced. A `Future` is a value representing some computation that will complete at
-some point in the future.
+with a value representing the work of creating a TCP stream.
 
-It's worth reiterating that returned futures are lazy, i.e., no work is performed when
-calling the combinator. Instead, once all the asynchronous steps are sequenced, the
-final `Future` (representing the entire task) is 'spawned' (i.e., run). This is when
-the previously defined work starts getting run. In other words, the code we've written
-so far does not actually create a TCP stream.
+Note the `async`-block that follows. Contents of the async block are not executed immediately but are rather
+stored in an object whose result can be computed later. Inside the `async` block we can `await` on the
+created future so that any code that follows is executed with the result of the `await`-ed [`Future`]'s computation.
 
 We will be digging more into futures (and the related concepts of streams and sinks)
 later on.
 
-It's also important to note that we've called `map_err` to convert whatever error
-we may have gotten to `()` before we can actually run our future. This ensures that
-we acknowledge errors.
-
 Next, we will process the stream.
 
 # Writing data
 
 Our goal is to write `"hello world\n"` to the stream.
 
-Going back to the `TcpStream::connect(addr).and_then` block:
+Going back to the `TcpStream::connect(addr)` block:
 
 ```rust
-# #![deny(deprecated)]
-# extern crate tokio;
-#
-# use tokio::io;
-# use tokio::prelude::*;
-# use tokio::net::TcpStream;
-# fn main() {
-# let addr = "127.0.0.1:6142".parse().unwrap();
-let client = TcpStream::connect(&addr).and_then(|stream| {
-    println!("created stream");
-
-    io::write_all(stream, "hello world\n").then(|result| {
-      println!("wrote to stream; success={:?}", result.is_ok());
-      Ok(())
-    })
-})
-# ;
-# }
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpStream;
+
+fn main() {
+    let addr = "127.0.0.1:6142";
+    let client = TcpStream::connect(addr);
+
+    let future = async move {
+        if let Ok(mut stream) = client.await {
+            println!("created stream");
+
+            let result = stream.write_all(b"hello world\n").await;
+
+            println!("wrote to stream; success={:?}", result.is_ok());
+        }
```

Comment:
Before we added the call to `write_all`, this code used a `match`, but now it uses an `if let`...why not continue using the match? That way, we still show error handling, and less code changes between the two examples.